high_school/views.py

```diff
@@ -3,61 +3,115 @@
-                    serializer.initial_data[x]["end_time"] = "N/A"
-            if serializer.is_valid():
-                response = serializer.save()
-                return render(request, "high_school/index.html", {"response": response})
-            else:
-                print(serializer.errors)
+            response = save_high_schools(limit)
+            if response:
+                save_programs(limit)
+            return render(request, "high_school/index.html", {"response": response})
+
     else:
         form = SaveHighSchoolsForm()
     return render(request, "high_school/save_high_schools.html", {"form": form})
 
 
+def save_high_schools(limit):
+    results = client.get(
+        ApiInfo.API_RESOURCE, select=ApiInfo.LOCAL_FIELD_LIST, limit=limit
+    )
+    serializer = HighSchoolSerializer(data=results, many=True)
+    for x in range(limit):
+        # for testing purposes, we should not be using actual high school emails
+        # all school emails are replaced with Patryk's email for future implementation of # noqa : E501
+        # supervisor email being taken directly from the school.
+        serializer.initial_data[x]["school_email"] = "pp2224@nyu.edu"
+        try:
+            # check if there is a start time in the school info
+            # if it is there then get the substring that ends it after "am"
+            # some fields have text after the time that is not necessary
+            if serializer.initial_data[x]["start_time"]:
+                am_loc = serializer.initial_data[x]["start_time"].find("am")
+                serializer.initial_data[x]["start_time"] = serializer.initial_data[x][
+                    "start_time"
+                ][: am_loc + 2]
+        except KeyError:
+            # if there is no start time provided in the info set it to N/A
+            serializer.initial_data[x]["start_time"] = "N/A"
+        try:
+            # check if there is a end time in the school info
+            # if it is there then get the substring that ends it after "pm"
+            # some fields have text after the time that is not necessary
+            if serializer.initial_data[x]["end_time"]:
+                pm_loc = serializer.initial_data[x]["end_time"].find("pm")
+                serializer.initial_data[x]["end_time"] = serializer.initial_data[x][
+                    "end_time"
+                ][: pm_loc + 2]
+        except KeyError:
+            # if there is no end time provided in the info set it to N/A
+            serializer.initial_data[x]["end_time"] = "N/A"
+    if serializer.is_valid():
+        return serializer.save()
+    else:
+        print(serializer.errors)
+        return None
+
+
+def extract_offer_rate(offer_rate_data):
+    if not offer_rate_data or offer_rate_data[0] != "-":
+        return 0
+    # Offer rate in the dataset is like:
+    # -96% of offers went to this group
+    return offer_rate_data.split("%")[0][1:]
+
+
+def parse_result(result):
```

Comment:
We'll need to add test cases for this function, but I'm not sure how to proceed with this because I'm trying to figure out a way to write test cases for the save_high_school function. If you have an idea, please do share.

Reply:
Yes, we should look into it in the upcoming sprints.